cpp/src/strings/case.cu

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/column/column.hpp>
+#include <cudf/column/column_factories.hpp>
+#include <cudf/column/column_device_view.cuh>
+#include <cudf/strings/strings_column_view.hpp>
+#include <cudf/strings/string_view.cuh>
+#include <cudf/strings/case.hpp>
+#include <cudf/utilities/error.hpp>
+#include "char_types/is_flags.h"
+#include "./utilities.hpp"
+#include "./utilities.cuh"
+
+#include <rmm/thrust_rmm_allocator.h>
+#include <thrust/transform.h>
+#include <thrust/transform_scan.h>
+
+namespace cudf
+{
+namespace strings
+{
+namespace
+{
+
+/**
+ * @brief Used as template parameter to divide size calculation from
+ * the actual string operation within a function.
+ * Useful when most of the logic is identical for both passes.
+ */
+enum TwoPass
```

Comment:
Just use a bool template parameter.

Reply:
I think naming these makes the code more clear where it is being used so I don't have go back up and figure out what true means for example.